torch/_inductor/decomposition.py

```diff
@@ -176,6 +176,9 @@ def special_erf(x):
     a5 = 1.061405429
     p = 0.3275911
 
+    if x.type == torch.bool:
```

Comment:
this is only half fix, if you look below, there's `torch.exp(-x*x)` that will be horribly wrong for uint8 argument (try it!), why doesn't OpInfo catch it?
The proper fix is to probably move this to lowering, correctly cast `x` to computation dtype (there's INT_TO_FLOAT type promotion kind in lowerings that should be used for that), and have dtype-specific lowering that'll call libdevice `erf` for double precision arguments because the accuracy of this approximation is not enough (OpInfo tests are actually failing with accuracy for this). 
Bonus points for comparing libdevice perf with this approximation for fp data dypes, and using libdevice always if it's fast enough.